kernel/init.c

```diff
@@ -302,7 +302,7 @@ static void bg_thread_main(void *unused1, void *unused2, void *unused3)
 void __weak main(void)
 {
 	/* NOP default main() if the application does not provide one. */
-	arch_nop();
```

Comment:
Here's a stupid question: why do we even have this function at all?  The default `main()` could just as easily be empty, right?  and `arch_nop()` isn't used anywhere else.

Reply:
IIRC it was something to do with MISRA-C, @ceolin would know.

Reply:
without this nop the code is considered dead code, since it is doing absolutely nothing and could be removed. Of course, we argue/document the reason of having it, but it seemed easier add this nop and silent static analysis tools